rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,3804 @@
+> solve this issue, you need to tell Git about the proxy:
+>
+> ```shell
+> $ git config --global http.proxy http://user:password@proxy.url
+> $ git config --global https.proxy http://user:password@proxy.url
+> ```
+>
+> When you connect to another network that doesn't use a proxy, you will need to
+> tell Git to disable the proxy using:
+>
+> ```shell
+> $ git config --global --unset http.proxy
+> $ git config --global --unset https.proxy
+> ```
+
+> **Password Managers**
+>
+> If your operating system has a password manager configured, `git push` will
+> try to use it when it needs your username and password.  For example, this
+> is the default behavior for Git Bash on Windows. If you want to type your
+> username and password at the terminal instead of using a password manager,
+> type:
+>
+> ```shell
+> $ unset SSH_ASKPASS
+> ```
+>
+> in the terminal, before you run `git push`.  Despite the name, [Git uses
+> `SSH_ASKPASS` for all credential
+> entry](https://git-scm.com/docs/gitcredentials#_requesting_credentials), so
+> you may want to unset `SSH_ASKPASS` whether you are using Git via SSH or
+> https.
+>
+> You may also want to add `unset SSH_ASKPASS` at the end of your `~/.bashrc`
+> to make Git default to using the terminal for usernames and passwords.
+
+Our local and remote repositories are now in this state:
+
+*TODO: Figure - GitHub Repository After First Push [../fig/github-repo-after-first-push.svg](https://github.com/swcarpentry/git-novice/blob/gh-pages/fig/github-repo-after-first-push.svg)*
+
+
+> **The '-u' Flag**
+>
+> You may see a `-u` option used with `git push` in some documentation.  This
+> option is synonymous with the `--set-upstream-to` option for the `git branch`
+> command, and is used to associate the current branch with a remote branch so
+> that the `git pull` command can be used without any arguments. To do this,
+> simply use `git push -u origin master` once the remote has been set up.
+
+We can pull changes from the remote repository to the local one as well:
+
+```shell
+$ git pull origin master
+```
+
+```text
+From https://github.com/vlad/planets
+ * branch            master     -> FETCH_HEAD
+Already up-to-date.
+```
+
+Pulling has no effect in this case because the two repositories are already
+synchronized.  If someone else had pushed some changes to the repository on
+GitHub, though, this command would download them to our local repository.
+
+> **Uploading files directly in GitHub browser**
+>
+> Github also allows you to skip the command line and upload files directly to 
+> your repository without having to leave the broswer. There are two options. 
+> First you can click the "Upload files" button in the toolbar at the top of the
+> file tree. Or, you can drag and drop files from your desktop onto the file 
+> tree. You can read more about this [on this GitHub page](https://help.github.com/articles/adding-a-file-to-a-repository/)
+
+
+
+## Collaborating {#rse-cmdline-git-collaborating}
+
```

Comment:
I wonder if this should be changed to a "collaborating with self" workflow because in real life it's pretty rare for multiple people to have push access to the same repo.